python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -517,3 +561,52 @@ def _private_display(self, reg_slice, this_format, default_width, markup):
             ds += markup(this_format % tuple(vals), colour)
 
         return ds
+
+    @staticmethod
+    def jobHasChildrenTest(jobDirectory, datafileName):
```

Comment:
That's not how function naming works. The name should reflect the API, not the implementation. If the purpose of this function is to answer the question "does this job have children" then the function should be named `jobHasChildren`. But even then, this seems like a strange place to ask this question; this class is always attached to a job object so the user should ask *that* object if it has children.

This function seems to answering the question above while also checking for the presence of the relevant directories (not raising an error if they're not found, simply printing a warning) but if there's no job directory then saying that the job has no children. That's a very confusing function to know how to use correctly.

If the API user wants to know of the job has children then they should ask *that*. if they want to know if there are any missing subjob directories then they should ask *that*. If they want to check for the presence of the job directory then they should check *that*. Combining all three into one function named `jobHasChildrenTest()` does not help with the complexity of the code.